datadriven/src/test_data_reader.rs

```diff
@@ -0,0 +1,153 @@
+use crate::line_sparser::parse_line;
+use crate::test_data::TestData;
+use anyhow::Result;
+use std::path::{Path, PathBuf};
+// use for writeln! macro
+use std::fmt::Write;
+use std::iter::Enumerate;
+use std::str::Lines;
+
+pub struct TestDataReader<'a> {
+    source_name: PathBuf,
+    pub data: TestData,
+    scanner: Enumerate<Lines<'a>>,
+    logger: slog::Logger,
+}
+
+impl<'a> TestDataReader<'a> {
+    pub fn new<P>(source_name: P, content: &'a str, logger: &slog::Logger) -> Self
+    where
+        P: AsRef<Path>,
+    {
+        Self {
+            source_name: source_name.as_ref().to_path_buf(),
+            scanner: content.lines().enumerate(),
+            data: TestData::default(),
+            logger: logger.clone(),
+        }
+    }
+
+    pub fn next(&mut self) -> Result<bool> {
+        loop {
+            let line = self.scanner.next();
+            if line.is_none() {
+                return Ok(false);
+            }
+
+            self.data = TestData::default();
+            self.data.pos = format!(
+                "{} : L{}",
+                self.source_name.as_path().display(),
+                line.unwrap().0
```

Comment:
Should be `line.unwrap().0 + 1`.

Reply:
The original implementation points line to the end of continuous line. For example,
```
a \
b \
c
```
The line number is 3 instead of 0.